src/MPI_with_SYCL/send_recv_buff.cpp

```diff
@@ -66,19 +68,27 @@ int main(int argc, char *argv[]) {
         auto kern = [=](sycl::id<1> id) { acc[id] *= 2; };
         h.parallel_for(sycl::range<1>{nelem}, kern);
       };
-      // When using buffers with CUDA-aware MPI, a host_task must be used with a
+      // When using buffers with GPU-aware MPI, a host_task must be used with a
       // sycl::interop_handle in the following way. This host task command group
       // uses MPI_Send to send the data to rank 1.
       auto ht = [&](sycl::handler &h) {
         sycl::accessor acc{buff, h};
         h.host_task([=](sycl::interop_handle ih) {
-          // get the native CUDA device pointer from the SYCL accessor.
-          auto cuda_ptr = reinterpret_cast<int *>(
-              ih.get_native_mem<sycl::backend::ext_oneapi_cuda>(acc));
+// get the native GPU device pointer from the SYCL accessor.
+#if defined(USE_HIP)
+          auto gpu_ptr = reinterpret_cast<int *>(
+              ih.get_native_mem<sycl::backend::hip>(acc));
+#elif defined(USE_L0)
+          auto gpu_ptr = reinterpret_cast<int *>(
+              ih.get_native_mem<sycl::backend::level_zero>(acc));
+#else
+          auto gpu_ptr = reinterpret_cast<int *>(
+              ih.get_native_mem<sycl::backend::cuda>(acc));
```

Comment:
Does this build for you? The [backend enum](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/backend_types.hpp) values are:
`ext_oneapi_hip`, `ext_oneapi_level_zero`, `ext_oneapi_cuda`

Reply:
I've fixed this now thanks. Apparently they are changing back to `cuda` etc but apparently not yet.